### PR TITLE
chore: upgrade snaps-controller and adapts its usage

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR upgrades the snap-controller version to its latest version.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -171,7 +171,7 @@
     "@metamask/signature-controller": "^14.0.0",
     "@metamask/slip44": "3.1.0",
     "@metamask/smart-transactions-controller": "10.0.0",
-    "@metamask/snaps-controllers": "^5.0.0",
+    "@metamask/snaps-controllers": "^8.1.1",
```
Comment (FrederikBolding) [inline package.json:174]: Can we bump `snaps-rpc-methods`, `snaps-sdk` and `snaps-utils` too? 🫡

Comment (Jonathansoufer) [reply package.json:174]: Yes we can 🫡!

**Review by tommasini — DISMISSED** [review]
LGTM !

**Review by owencraston — COMMENTED** [review]
Something weird is going on with this version of the snaps controller. I tried to click into the type definition of `SnapsAllowedActions` and it simply brings me to this

Screenshot 2024-06-05 at 3 48 19 PM

**Comment by Jonathansoufer** [comment]
> SnapsAllowedActions

That was me before a change made on snaps controller. Removed!

**Review by owencraston — DISMISSED** [review]

**Review by tommasini — APPROVED** [review]
LGTM from a platform perspective! Will defer to @FrederikBolding on snaps perspective since I miss context there!

**Review by owencraston — APPROVED** [review]

File: babel.config.js [hunk 2]
```diff
@@ -6,6 +6,12 @@ module.exports = {
     'transform-inline-environment-variables',
     'react-native-reanimated/plugin',
   ],
+  overrides: [
+    {
+      test: './node_modules/marked',
```
[the remark this answers is not in the document]
Comment (Jonathansoufer) [reply babel.config.js:11]: yep!

**Pull request merged**